compiler/types.nim

```diff
@@ -1183,8 +1183,8 @@ proc typeAllowedAux(marker: var IntSet, typ: PType, kind: TSymKind,
     if result.isNil and t.sons[0] != nil:
       result = typeAllowedAux(marker, t.sons[0], skResult, flags)
   of tyTypeDesc:
-    # XXX: This is still a horrible idea...
-    result = nil
+    # Could be inside a macro, so var and let are allowed here and checked later
+    if kind notin {skParam, skResult, skVar, skLet}: result = t
```

Comment:
Someone might know how to fix the concept tests without reverting this, so I'm keeping it for now unless someone thinks it should be reverted or if it doesn't get merged